client/src/pages/CalendarPage/CalendarPage.js

```diff
@@ -32,23 +36,70 @@ const dailyTaskDateFormatter = (date) => {
 };
 
 const CalendarPage = () => {
+  const [currDate, setCurrDate] = useState(new Date());
+  const [taskData, setTaskData] = useState([]);
+  const [toggleError, setToggleError] = useState(false);
+  const [loadingError, setLoadingError] = useState(false);
+  const [toggleErrorMessage, setToggleErrorMessage] = useState("");
+  const [loadingErrorMessage, setLoadingErrorMessage] = useState("");
+  const filterSet = ["Not Started", "Ongoing", "Completed"];
+  const [currentFilter, setCurrentFilter] = useState(filterSet[0]);
+
+  const filterTask = (task) => {
+    switch (currentFilter) {
+      case filterSet[0]: // not started
+        return !task.isStarted;
+      case filterSet[1]: // ongoing
+        return task.isStarted && !("taskEndedAt" in task);
+      case filterSet[2]:
+        return task.isStarted && "taskEndedAt" in task;
+    }
+  };
+
+  const getTasks = async () => {
+    setLoadingErrorMessage("");
+    setLoadingError(false);
+    await instance
+      .get(buildDateRangeRoute(currDate, currDate))
+      .then((taskData) => {
+        const filteredTaskData = taskData.data.filter(filterTask);
+        if (filteredTaskData.length === 0) {
+          setLoadingErrorMessage("No tasks yet");
+          setLoadingError(true);
+        } else {
```

Comment:
We can remove the else because we already setLoadingError to be false at the beginning of the function and if the `if` portion of the statement doesn't execute, then it will remain as false